scripts/nb-tester/qiskit_docs_notebook_tester/config.py

```diff
@@ -59,143 +60,166 @@ def get_notebook_jobs(args: argparse.Namespace) -> Iterator[NotebookJob]:
-                    "control_flow": config.args.control_flow
-                }
+    for group in config.groups:
+        for path in group["notebooks"]:
+            if config.cli_filenames and not matches(path, config.cli_filenames):
+                continue
+            if config.test_strategy not in group.get("test-strategies", {}):
+                if matches(path, config.cli_filenames):
+                    # User explicitly defined this path so we'll explain why we're skipping it
+                    print(
+                        f'ℹ️ Skipping {path} (no test strategy "{config.test_strategy}" defined for this notebook)'
+                    )
+                continue
+
+            patch = config.get_patch_for_group(group)
+
+            if patch:
+                write = Result(False, "hardware was mocked")
+            elif not config.write:
+                write = Result(False, "--write arg not set")
+            else:
+                write = Result(True)
+
+            yield NotebookJob(
+                path=Path(path),
+                pre_execute_code=PRE_EXECUTE_CODE,
+                backend_patch=patch,
+                cell_timeout=config.cell_timeout,
+                write=write,
             )
-        
-        write = Result(True)
-        if reason := config.should_skip_writing(path):
-            write = Result(False, reason)
-
-        yield NotebookJob(
-            path=path,
-            pre_execute_code=PRE_EXECUTE_CODE,
-            backend_patch=backend_patch,
-            cell_timeout=-1 if config.args.submit_jobs else 300,
-            write=write,
-        )
 
 
 @dataclass
 class Config:
-    args: argparse.Namespace
-    notebooks_normal_test: list[str]
-    notebooks_exclude: list[str]
-    notebooks_that_submit_jobs: list[str]
-    notebooks_no_mock: list[str]
-
-    @property
-    def all_job_submitting_notebooks(self) -> list[str]:
-        return [*self.notebooks_that_submit_jobs, *self.notebooks_no_mock]
-
-    @property
-    def all_notebooks_to_test(self) -> list[str]:
-        return [
-            *self.notebooks_normal_test,
-            *self.notebooks_that_submit_jobs,
-            *self.notebooks_no_mock,
-        ]
-
-    @property
-    def all_notebooks(self) -> list[str]:
-        return [
-            *self.all_notebooks_to_test,
-            *self.notebooks_exclude,
-        ]
+    cli_filenames: list[Path]
+    cell_timeout: int
+    test_strategy: str
+    write: bool
+    groups: list[dict]
 
     @classmethod
     def from_args(cls, args: argparse.Namespace) -> Config:
         """
-        Create config from args, including loading the globs from the TOML file
+        Create config from args, including loading the paths from the TOML file if needed.
         """
-        path = Path(args.config_path)
+        if (args.config_path or args.test_strategy) and args.patch:
+            raise ValueError("Can't set --patch with --config-file or --test-strategy")
```

Comment:
I think it's okay to keep this check here with the custom message, but I recently found that we could use the mutual exclusion feature from `argparse` for these kinds of things: https://docs.python.org/3/library/argparse.html#mutual-exclusion